docker-compose.yaml

```diff
@@ -0,0 +1,419 @@
+version: '2.4'
+services:
+  nginx:
+    container_name: nginx
+    image: 'index.docker.io/library/nginx:1.17.7@sha256:8aa7f6a9585d908a63e5e418dc5d14ae7467d2e36e1ab4f0d8f9d059a3d071ce'
+    volumes:
+      - './nginx:/etc/nginx'
+    ports:
+      - '0.0.0.0:7080:7080'
+    networks:
+      - sourcegraph
+    restart: always
+
+  # Description: Serves the frontend of Sourcegraph via HTTP(S).
+  #
+  # Disk: 128GB / non-persistent SSD
+  # Network: 100mbps
+  # Liveness probe: HTTP GET http://sourcegraph-frontend:3080/healthz
+  # Ports exposed to other Sourcegraph services: 6060/TCP
+  # Ports exposed to the public internet: 3080 (HTTP) and/or 3443 (HTTPS)
+  #
+  # Note: SRC_GIT_SERVERS, SEARCHER_URL, and SYMBOLS_URL are space-separated
+  # lists which each allow you to specify more container instances for scaling
+  # purposes. Be sure to also apply such a change here to the frontend-internal
+  # service.
+  sourcegraph-frontend-0:
```

Comment:
All of the services in this file need to have the `--cpus` and `--memory` copied over from the sh scripts

Reply:
@slimsag Note, adding the `cpus` flags means that I can no longer run docker-compose up on my machine since it only has two cores.

Reply:
This is expected; if you have less cores than is required you'll need to lower the CPU counts yourself. The docker-compose file should be optimized for a prod deployment setting initially I think